CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+
+## Development Guidelines
+
+In this section we are going to describe the coding rules for contributing to the raiden repository. All code you write should strive to comply with these rules.
+
+### Coding Style
+
+#### Python
+
+Raiden is written in Python and we follow the official Python style guide [PEP8](https://www.python.org/dev/peps/pep-0008/). It is highly
+recommended to use the [flake8](https://pypi.python.org/pypi/flake8) tool in order to automatically determine any and all style violations. The customizeable part of flake can be seen in the [configuration file](setup.cfg). For all the rest which are not
+configurable here is some general guidelines.
+
+** Line Length **
+Flake8 will warn you for 99 characters which is the hard limit on the max length. Try not to go above it. We also have a soft
+limit on 80 characters but that is not enforced and is there just to encourage short lines.
+
+** Breaking function definitions when line is above 99 characters **
+
+Always put each argument into its own line. Look at the following examples to understand:
+
+The following should be avoided
+```python
+
+def function_with_many_args(argument1, argument2, argument3, argument4, argument5, argument6, argument7):
+	pass
+```
+
+and instead you should
+
+```python
+
+def function_with_many_args(
+    argument1,
+    argument2,
+    argument3,
+    argument4,
+    argument5,
+    argument6,
+    argument7):
+	pass
+```
+
+
+** Breaking function calls when line is above 99 characters **
+
+Much like in the above example the following should be avoided
+
+```python
+
+function_call_with_many_arguments(argument1, argument2, argument3, argument4, argument5, argument6, argument7)
+
+```
+
+and instead you should
+
+```python
+
+function_call_with_many_arguments(
+    argument1,
+    argument2,
+    argument3,
+    argument4,
+    argument5,
+    argument6,
+    argument7
+)
+```
+
+Difference being that you can place the closing parentheses in the next line.
+
+
+**Naming Style**
+
+Use [Snake Case](https://en.wikipedia.org/wiki/Snake_case) for variable and function names and [Camel Case](https://en.wikipedia.org/wiki/Camel_case) for class names.
+
+**Naming Convention**
+
+Use descriptive variable names and avoid short abbreviations.
+
+
+The following is bad:
+```python
+mgr = Manager()
+a = AccountBalanceHolder()
+s = RaidenService()
+```
+
+While this is good:
+
+```python
+manager = Manager()
+balance_holder = AccountBalanceHolder()
+service = RaidenService()
+```
+
+#### Solidity
+
+For solidity we generally follow the style guide as shown in the [solidity documentation](http://solidity.readthedocs.io/en/develop/style-guide.html)
+with a few notable exceptions:
+
+** Variable Names **
+
+All variable name should be in snake case, just like in python. Function names on the other hand stay in CamelCase.
```

Comment:
Okay. So camel case can be both. mixedCase seems to be a python term, that is the definition of what we want.